src/pyscaffoldext/custom_extension/custom_extension.py

```diff
@@ -149,3 +157,24 @@ def add_custom_extension_structure(struct, opts):
                             helpers.NO_OVERWRITE)
 
     return struct, opts
+
+
+def check_project_name(struct, opts):
+    """
+    Enforce the naming convention of PyScaffold extensions.
+    The project name must start with 'pyscaffoldext-' and
+    the package name shouldn't contain the redundant
+    'pyscaffoldext_' in the beginning of the name.
+    :param struct:
+    :param opts:
+    :return:
+    """
+    if not opts['project'].startswith('pyscaffoldext-'):
```

Comment:
Maybe we should still allow to do whatever the user wants in case ``--force`` is provided.